### PR TITLE
Make DataModel.update no longer update from extra_fits

### LINKED ISSUE #4591: Datamodels.update() should not update from `extra_fits`.
Issue [JP-1310](https://jira.stsci.edu/browse/JP-1310) was created by James Davies:

It seems that FITS keywords that are not in the schema and are passed along to `extra_fits` get copied over via `model.update()` when a new datamodel is updated from an old one that has `extra_fits`.  This is not desirable.  Only keywords in the schema should be copied over.

Also maybe the default `DataModel.save` should not write out anything from `extra_fits`, and the user should be explicit here.  Something like `model.save(filename, extra_fits=True)`.

### PR DESCRIPTION
**Pull request by jdavies-st**
DataModel.update() method will now ignore `extra_fits` in the model that is providing the updated metadata, unless requested via keyword arg `extra_fits=True`.

Resolves #4591

### CODE REVIEW

File: jwst/datamodels/model_base.py [hunk 1]
```diff
@@ -948,12 +948,6 @@ def protected_keyword(path):
             if not protected_keyword(path):
                 set_hdu_keyword(self._instance, d, path)
 
-        # Perform updates to extra_fits area of a model
```
Comment (hbushouse) [inline jwst/datamodels/model_base.py]: I would really like to see this controlled by an optional argument, rather than just eliminating the capability.

**Comment by jdavies-st** [comment]
One thing I'm not entirely clear on in the `update()` method API is what `only` arg does.  The documentation is bad, and the default value of `""` is, um, not good.  @hbushouse do you have any insight what its purpose is?  I feel like there could be an odd interaction between `only` and `extra_fits` args.

**Comment by hbushouse** [comment]
IIRC, the "only" arg was used to copy/update only meta data from the primary header (no extensions).

**Comment by jdavies-st** [comment]
I think the tests now should cover the combos of `only` and `extra_fits`.  And it seems to behave how we expect it to.

I also found a test for this method over in `test_fits.py`.  That test was in the wrong location and was very simple and not very intuitive.  I removed it, and added another test in `test_models.py`.  I think the new tests cover all the use cases and give good coverage to the codebase.

**Pull request merged**